### PR TITLE
Display dorms on an embedded Google map

### PR DESCRIPTION
**Pull request by ZiqiXiong**
This PR replaces the table of latest reviews with a map that displays all the dorms and their ratings on the homepage of our Housing app. 

screenshot 2016-09-10 15 21 15

### CODE REVIEW

File: aspc/housing/views.py [hunk 1]
```diff
@@ -17,12 +17,12 @@
...
-    template_name = "housing/home.html"
-    date_field = "create_ts"
-    allow_empty = True
-    queryset = Review.objects.all().select_related('room')
-    paginate_by = 15
+def home(request):
+    if request.method == 'GET':
+        return render(request, 'housing/home.html')
+    dorms = Building.objects.filter(type=0, position__isnull=False)
+    dorm_objects = [dorm.map_object() for dorm in dorms if dorm.map_object()]
```
Comment (KentShikama) [inline aspc/housing/views.py]: Use `filter`?

File: aspc/static/js/dorm_map.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+          $.ajax({
+              url: '',
+              type: 'POST',
+              data: {
+                csrfmiddlewaretoken: csrftoken
+              },
+              success: function (data) {
+                  for (var i in data) {
+                      console.log(data[i]);
```
Comment (KentShikama) [inline aspc/static/js/dorm_map.js]: Remove

File: aspc/static/js/dorm_map.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    // Does this cookie string begin with the name we want?
+                    if (cookie.substring(0, name.length + 1) === (name + '=')) {
+                        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                        break;
+                    }
+                }
+            }
+            return cookieValue;
+        }
+        var csrftoken = getCookie('csrftoken');
```
Comment (KentShikama) [inline aspc/static/js/dorm_map.js]: Extract to https://www.npmjs.com/package/cookies-js

File: aspc/college/models.py [hunk 4]
```diff
@@ -159,6 +162,39 @@ def get_map_info(self):
         }
         return data
 
+    def get_average_ratings(self):
+        Room = apps.get_model('housing', 'Room')
+        rooms = Room.objects.filter(floor__building=self, average_rating__isnull=False)
+        room_count = float(rooms.count())
+        if not room_count:
+            return {
```
Comment (KentShikama) [inline aspc/college/models.py]: Extract method

File: aspc/college/models.py [hunk 5]
```diff
@@ -159,6 +162,39 @@ def get_map_info(self):
...
+                'average_rating' : 0,
+                'average_rating_quiet' : 0,
+                'average_rating_spacious' : 0,
+                'average_rating_temperate' : 0,
+                'average_rating_maintained' : 0,
+                'average_rating_cellphone' : 0
+            }
+
+        return {
+            'average_rating' : sum(rooms.values_list('average_rating', flat=True)) / room_count + 1,
```
Comment (KentShikama) [inline aspc/college/models.py]: Extract method for 

```
def name(rooms, room_count, string):
    return sum(rooms.values_list(string, flat=True)) / room_count + 1
```

File: aspc/housing/templates/housing/dorm_map.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<div style='height:500px;' id="map"></div>
```
Comment (KentShikama) [inline aspc/housing/templates/housing/dorm_map.html]: Extract inline css

**Comment by ZiqiXiong** [comment]
@KentShikama  Fixed

**Pull request merged**